### PR TITLE
test: Setup e2e tests in CI

### PR DESCRIPTION
**Pull request by grzesuav**
Added e2e tests for k8s 1.16-1.19, with exluded `catset` (due to https://github.com/metacontroller/metacontroller/issues/144)

Before adding k8s 1.13-1.15 examples  need to be refactoreda bit (as their crd's are in v1 version) - some work is needed to make it work with bot v1 and v1beta1 versions.

### CODE REVIEW

**Comment by grzesuav** [comment]
also, I want to discover if feasible would be to collect coverage from integration tests

File: .github/workflows/pipeline.yaml [hunk 1]
```diff
@@ -6,27 +6,114 @@ jobs:
...
+        kind load docker-image metacontrollerio/metacontroller:dev
+    - name: Apply metacontroller manifests
+      run: |
+        kubectl apply -k manifests/dev
+        sleep 20s
+        kubectl get pods -n metacontroller
+    - name: Run e2e tests
+      run: |
+        cd examples
+        ./test.sh catset
```
Comment (AmitKumarDas) [inline .github/workflows/pipeline.yaml:109]: Is it possible to provide catset as part of arg.
I took sometime to understand that catset is actually excluded as part of this test

**Review by AmitKumarDas — COMMENTED** [review]
:heart: A good one @grzesuav 
Do you think it will be good to remove the commented code/yaml

File: .github/workflows/pipeline.yaml (hunk 1 above)
Comment (grzesuav) [reply .github/workflows/pipeline.yaml:109]: yeah, it was quick hack ;) I can try to make it proper posarg

**Comment by grzesuav** [comment]
>  A good one @grzesuav
> Do you think it will be good to remove the commented code/yaml

hmm let me think a bit how to do it, I wanted to have it in some form to not loose it

**Comment by grzesuav** [comment]
@AmitKumarDas reorganized code to be more concise

**Review by AmitKumarDas — APPROVED** [review]

**Pull request merged**